docs/route-tasks.md

```diff
@@ -0,0 +1,171 @@
+
+```go
+// Worker is created and subscribed to the "hello-world" Task Queue
+w := worker.New(c, "hello-world", worker.Options{})
+defer w.Stop()
+// Worker is registered to handle the helloworld Workflow
+w.RegisterWorkflow(helloworld.Workflow)
+// Worker is registered to handle the helloworld Activity
+w.RegisterActivity(helloworld.Activity)
+```
+
+In the next code snippet, we [start the `helloworld.Workflow`](https://github.com/temporalio/go-samples/blob/master/helloworld/starter/main.go#L25):
+
+```go
+// Workflow Options specify the "hello-world" Task Queue
+// Which is the Task Queue that the Workflow will send its Tasks to
+workflowOptions := client.StartWorkflowOptions{
+  ID:        "hello_world_workflowID",
+  TaskQueue: "hello-world",
+}
+// Workflow Options are passed to the execution of the Workflow
+we, err := c.ExecuteWorkflow(
+  context.Background(), 
+  workflowOptions, 
+  helloworld.Workflow, 
+  "Temporal",
+)
+```
+
+:::caution
+
+If a Worker pulls a Task that it has not been registered to run, an "unknown workflow type" error will occur.
+
+:::
+
+## Task routing for Activity dependencies
+
+The Temporal service will persist Activity results as long as the results are less than 2MB in size and can be serialized for storage (which happens to be true for most Activity results). However, there are some scenarios where an Activity execution will do something that results in data, such as a large file, being stored directly on the host where the Worker is running. It is also possible that the next Activity in the Workflow has a dependency on that data and needs to run on the same host.
+
+Let's look at a file processing Workflow example where files are downloaded, processed in some way, and then uploaded to some other location. If a large video file is being processed only the name of the file can persist normally. The file itself must persist locally on the Worker host, as it is too big for normal Temporal storage. This means that any future Activities that act on the video file must also take place on the host where the Worker downloaded it.
+
+So, how do we ensure that a Worker on the correct host will execute the next Activity that depends on the video file?
+
+This is accomplished in different ways depending on the SDK.
+
+### Java
+
+Consider the [file processing sample in Java](https://github.com/temporalio/java-samples/tree/master/src/main/java/io/temporal/samples/fileprocessing). The paradigm is to create two Workers. Each will run on the same host and thus have access to the same file system, but one will listen to a global (shared) Task Queue, which it will use to start the Workflow, and the other will listen to its own host-specific (private) Task Queue. 
```

Comment:
This is a very good explanation. It might be worth briefly commenting on how "host-specific (private)" is accomplished (unique host id/mac address etc)